pkg/util/policy/agent/agent.py

```diff
@@ -78,48 +70,92 @@ def route(self):
+
+        return
+
+
     def do_NP_update(self):
-        method = self.json_data.get('method')
-        policy_def = self.json_data.get('policy_definition')
+        """
+        Installs or deinstalls network policy rules from iptables
+        """
+        global addr_scheme
 
-        if method not in [ 'ADDED', 'DELETED' ] or not policy_def:
-            # HTTP 422 - Unprocessable Entity seems to be relevant. We have verified that json is valid
-            # but expected fields are missing
-            self.send_response(HTTP_Unprocessable_Entity)
-            self.wfile.write("""Expected { "method" : "ADDED|DELETED", "policy_definition" : "NP" } """)
+        # Namespace isolation update is a special case
+        if self.is_ns_isolation(self.json_data):
+            self.do_NS_update(self.json_data)
+            return
+
+        policy_def = self.json_data
+        # if policy_def:
+        #     policy_def_valid = self.validate_policy(policy_def)
+        # else:
+        #     policy_def_valid = self.json_data.get("name")
+        #     policy_def = self.json_data
+        policy_def_valid = self.validate_policy(policy_def)
+        
+        logging.warning("In do_NP_update : policy_def_valid = %s" % policy_def_valid)
+
+        if policy_def_valid:
+            addr_scheme = policy_def.get("datacenter")
 
-        elif self.json_data['method'] == 'ADDED':
             self.send_response(200)
             self.wfile.write("Policy definition accepted")
             logging.warning("Creating policy %s" % policy_def)
-            policy_update(addr_scheme, policy_def)
 
-        elif self.json_data['method'] == 'DELETED':
-            self.send_response(200)
-            self.wfile.write("Policy definition deleted")
-            logging.warning("Deleting policy %s" % policy_def)
-            policy_update(addr_scheme, policy_def, delete_policy=True)
+            if self.http_method == "DELETE":
+                logging.warning("Deleting policy %s" % policy_def)
+            else:
+                logging.warning("Creating policy %s" % policy_def)
+
+            policy_update(addr_scheme, policy_def, delete_policy = ( self.http_method == "DELETE" ))
+
+            return
+        else:
+            self.send_response(HTTP_Unprocessable_Entity)
+            self.wfile.write(""" Failed to validate policy definition """)
+
 
-        return
 
 
-    def do_NS_update(self):
+    def do_NS_update(self, policy_def):
         # TODO This endpoint shoud really go away, only used for prototyping
         # while new policy fromat being discussed.
-		logging.info("Request hit /ns-update")
+        # TODO Now it will take a lot of refactoring to support for empty segment_ids
+        # which are required to express network policy that selects all segments.
+        # But still must be done.
+		logging.info("Request for NS isolation update")
 
-		tid = self.json_data.get('tid')
+                applied_to = policy_def.get('applied_to')
```

Comment:
Indentation in some of these lines is off. Looks like some discrepancy between TAB characters and spaces.

Can we please expand all tabs to spaces to avoid this sort of thing?